includes/admin/class-wc-payments-admin-sections-overwrite.php

```diff
@@ -18,11 +18,23 @@ class WC_Payments_Admin_Sections_Overwrite {
 	 */
 	private $previous_current_section;
 
+	/**
+	 * WC_Payments_Account instance to get information about the account.
+	 *
+	 * @var WC_Payments_Account
+	 */
+	private $account;
+
 	/**
 	 * WC_Payments_Admin_Sections_Overwrite constructor.
+	 *
+	 * @param WC_Payments_Account $account WC_Payments_Account instance.
 	 */
-	public function __construct() {
+	public function __construct( WC_Payments_Account $account ) {
+		$this->account = $account;
```

Comment:
Initially I was hoping to be able to skip all the `add_action` in case the account was not connected, but unfortunately we need to check for the account's connection when the action is running instead.

Reply:
If this class doesn't need to be instantiated when it is disconnected, what if we don't import the class at all here: 
https://github.com/Automattic/woocommerce-payments/blob/a2cfe06221eebbae39d1ab6241bf178eda843855/includes/class-wc-payments.php#L276-L277

Something like
```php
if ( connected ) {
  include_once __DIR__ . '/admin/class-wc-payments-admin-sections-overwrite.php';
  WC_Payments_Admin_Sections_Overwrite( self::get_account_service() );
}
```

Reply:
Some of the methods of the class do need to be instantiated.
Unfortunately tho, if we did something like `if ( $is_connected ) { /* import class /* }` we could still have an issue with the account data.

When I tried something similar ( 3a4d716198d5cfa8be269d966f9b253a7f316b9f ), the functional tests fail (and the settings stop working) because the cached account data is instantiated too soon, resulting in a permanent connection error.

Reply:
Oh look at that commit!

I see. The connection will always be thought as "not ready" since it was instantiated too soon in the cycle. Unless it checks and refreshes its status again, otherwise it stays as close forever. Using hooks inside each function makes sense.

Reply:
Yeah, I was also _very_ saddened about checking the status within each function, not gonna lie...